src/javascript/class-scanner.ts

```diff
@@ -574,3 +585,93 @@ class CustomElementsDefineCallFinder implements Visitor {
     return null;
   }
 }
+
+export function extractPropertiesFromConstructor(
+    astNode: estree.Node,
+    document: JavaScriptDocument,
+    properties = new Map<string, ScannedProperty>()) {
+  if (astNode.type === 'ClassExpression' ||
+      astNode.type === 'ClassDeclaration') {
+    for (const method of astNode.body.body) {
+      if (method.type !== 'MethodDefinition' || method.kind !== 'constructor') {
+        continue;
+      }
+      const constructor = method;
+      for (const statement of constructor.value.body.body) {
+        if (statement.type !== 'ExpressionStatement') {
+          continue;
+        }
+        let name;
+        let astNode;
+        let defaultValue;
+        if (statement.expression.type === 'AssignmentExpression') {
+          name = getPropertyNameOnThisExpression(statement.expression.left);
+          astNode = statement.expression.left;
+          defaultValue = escodegen.generate(statement.expression.right);
+        } else if (statement.expression.type === 'MemberExpression') {
```

Comment:
can you document his branch? I assume you looking for just `/** @public */ this.foo;`